rpm dependency on slurm, release rpm for multiple slurm versions

simonpintarelli opened the pull request:
- add `Require: slurm = SLURM_VERSION` to rpm spec
- makefile takes an option `RPM_SLURM_VERSION` (used only in rpm spec)
- add a script to make create and extract the binary rpm from the container
- Release rpm artifacts for multiple slurm versions

simonpintarelli committed:
enforce slurm version in rpm

- makefile uses `srun --version` to determine the slurm version (the slurm in
the container is handbuilt without rpm)

simonpintarelli committed:
add script to create binary rpm

simonpintarelli committed:
artifacts / slurm-version

bcumming commented on docker/Readme.md:
````diff
@@ -43,3 +43,8 @@ When making changes in the local source tree, the plugin can be rebuilt/resintal
+
+Build the binary rpm, e.g. for installation on production system:
+```bash
+docker compose exec -T slurm bash < make-rpm.sh
````
Should this be: `< make-binary-rpm.sh`?

simonpintarelli replied:
I'm sorry, I forgot to push the last commit. It is fixed now.

bcumming commented on .github/workflows/artifacts.yaml:
```diff
@@ -23,35 +23,34 @@ jobs:
+          make RPM_SLURM_VERSION=${SLURM_VER} rpm
           sudo rpmbuild --rebuild "$(find rpm/SRPMS -name '*.rpm' -type f -print -quit)" --define "_topdir $(pwd)/rpm"
-          sudo rpm --install "$(find rpm/RPMS -name '*.rpm' -type f -print -quit)"
+          # TODO: this fails, because the new rpm lists slurm as dependency (image installed slurm manually not via pkg manager)
```
Is this still the case?
If it is, maybe don't mark SLURM as required in the RPM.

bcumming replied:
Is SLURM installed as an RPM on the Alps clusters?

simonpintarelli replied:
Yes, on ALPS slurm is installed as RPM. But not in the container.

simonpintarelli replied:
I've added `rpm --install --nodeps` in the line above. Or shall we remove the explicit dependency on slurm from the rpm package?

jpcoles replied:
I added slurm as a dependency after we had the issue that the path for the plugin conf file did not exist if this package is installed before slurm. The dependency forces the installation order to be correct. Otherwise we have to explicitly create the directory when the rpm is installed, or address the issue in some other way.

bcumming commented on .github/workflows/artifacts.yaml:
```diff
@@ -13,8 +13,8 @@ jobs:
           - 20.11.9
-          - 21.08.8-2
-          - 22.05.5
+          # - 21.08.8-2
```
Remove this line?

simonpintarelli replied:
Done

simonpintarelli committed:
make-binary-rpm use same naming scheme as in ci

simonpintarelli committed:
ci: test install rpm

rpm --install --nodeps option since slurm is not installed via rpm in container
